virttest/utils_disk.py

```diff
@@ -182,6 +190,571 @@ def clean_old_image(image):
+                 show in 'diskpart list disk'.
+                 call function: get_windows_disks_index()
+    :param timeout: time for cmd execution
+    :return: True or False
+    """
+    detail_cmd = ' echo detail disk'
+    detail_cmd = _wrap_windows_cmd(detail_cmd)
+    set_rw_cmd = ' echo attributes disk clear readonly'
+    set_rw_cmd = _wrap_windows_cmd(set_rw_cmd)
+    online_cmd = ' echo online disk'
+    online_cmd = _wrap_windows_cmd(online_cmd)
+    for did in dids:
+        logging.info("Detail for 'Disk%s'" % did)
+        details = session.cmd_output(detail_cmd % did)
+        if re.search("Read.*Yes", details, re.I | re.M):
+            logging.info("Clear readonly bit on 'Disk%s'" % did)
+            status, output = session.cmd_status_output(set_rw_cmd % did,
+                                                       timeout=timeout)
+            if status != 0:
+                logging.error("Can not clear readonly bit: %s" % output)
+                return False
+        pattern = "DISK %s.*Offline" % did
+        if re.search(pattern, details, re.I | re.M):
+            logging.info("Online 'Disk%s'" % did)
+            status, output = session.cmd_status_output(online_cmd % did,
+                                                       timeout=timeout)
+            if status != 0:
+                logging.error("Can not online disk: %s" % output)
+                return False
+    return True
+
+
+def create_partition_table_linux(session, did, labeltype):
+    """
+    Create partition table on disk in linux guest.
+
+    :param session: session object to guest.
+    :param did: disk kname. e.g. sdb
+    :param labeltype: label type for the disk.
+    """
+    mklabel_cmd = 'parted -s "/dev/%s" mklabel %s' % (did, labeltype)
+    session.cmd(mklabel_cmd)
+
+
+def create_partition_table_windows(session, did, labeltype):
+    """
+    Create partition table on disk in windows guest.
+    mbr is default value, do nothing.
+
+    :param session: session object to guest.
+    :param did: disk index which show in 'diskpart list disk'.
+    :param labeltype: label type for the disk.
+    """
+    if labeltype == PARTITION_TABLE_TYPE_GPT:
+        mklabel_cmd = ' echo convert gpt'
+        mklabel_cmd = _wrap_windows_cmd(mklabel_cmd)
+        session.cmd(mklabel_cmd % did)
+
+
+def create_partition_table(session, did, labeltype, ostype):
+    """
+    Create partition table on disk in linux or windows guest.
+
+    :param session: session object to guest.
+    :param did: disk kname or disk index
+    :param labeltype: label type for the disk.
+    :param ostype: linux or windows.
+    """
+    if ostype == "windows":
+        create_partition_table_windows(session, did, labeltype)
+    else:
+        create_partition_table_linux(session, did, labeltype)
+
+
+def create_partition_linux(session, did, size, start, part_type=PARTITION_TYPE_PRIMARY):
+    """
+    Create single partition on disk in linux guest.
```

Comment:
There are some similar APIs in utils_misc.py for linux/windows w.r.t disk operations similar to that you have implemented.

I would request to review, refactor and place them in utils_disk.py.

**example:**
https://github.com/avocado-framework/avocado-vt/blob/56129a616849882ca0b35a4e6309abe379ae0819/virttest/utils_misc.py#L2960-L2975

Reply:
The related functions that you mentioned in utils_misc.py will be replaced by the new implementation in utils_disk.py. We move format disk related functions from utils_misc.py to utils_disk.py, and update them.